From 873725cc75b5fc0056ad3d0ee80e51f2cc0d041d Mon Sep 17 00:00:00 2001
From: Sunil Mohan Adapa <sunil@medhas.org>
Date: Wed, 12 Dec 2018 13:52:11 -0800
Subject: [PATCH 2/2] test: Fix case that checks if a db is installed

The test relied on mysql, sqlite, sqlite3 and find binaries not being found in
/bin, the path set during test. Since /bin is now a symlink to /usr/bin on many
machines, these binaries are found and test case fails.

Set the path to empty to make sure 'which' command is found and 'find' and other
database commands are not found. 'which' is a shell builtin from some shells
only.
---
 test/common | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/test/common b/test/common
index c809bf2..cdddef2 100755
--- a/test/common
+++ b/test/common
@@ -565,8 +565,11 @@ dbc_common_db_installed(){
     # the logic there is different and instead it fails on being unable
     # to find /usr/bin/find
     log_tc "${test_dbtype}d not in path"
-    ( PATH="/bin"; dbc_${test_dbtype}_db_installed ) > /dev/null 2>&1
+    # Copy which command needed for checking mysql, sqlite, sqlite3
+    cp /bin/which ./tmp/
+    ( PATH="./tmp"; dbc_${test_dbtype}_db_installed ) > /dev/null 2>&1
     assertFalse "${test_dbtype}d should not have been found" $?
+    rm -f ./tmp/which
 }
 
 dbc_common_escape_str(){
-- 
2.19.2

